subsys/zbus/Kconfig

```diff
@@ -9,6 +9,10 @@ menuconfig ZBUS
 
 if ZBUS
 
+config ZBUS_PREFER_DYNAMIC_ALLOCATION
+	bool "Set zbus to work with dynamic allocation using the system heap"
+	default y
```

Comment:
I'm hesitant to agree that this should be the default.

Reply:
Only advanced users, with proper knowledge and clear requirements, should use the zbus static version. When using the static version of zbus, the user must deal with several details needed only for critical or highly-constrained applications.

Any usual application, with safe wireless connectivity like WiFi or Bluetooth, should already use the heap, by the way. Most zbus users also use message subscribers that rely on the heap by default. So, I cannot see a reason not to use that. It would only bring traction to starting to use the subsystem. 

Again, if some application requires static-only memory, zbus is ready. The only thing I am trying to avoid is exposing too much complexity with no apparent reason.

Reply:
Bluetooth does not utilize the system heap, so I wouldn't use that as a reason for `ZBUS_PREFER_DYNAMIC_ALLOCATION` being the default. I am quite sure of this as it was working with zbus in Bluetooth applications that initially led me to discover the heap dependency. Bluetooth uses the static memory allocation pattern (which you say is only for advanced users :)) everywhere.

The only code in-tree that requires the heap (based on `HEAP_MEM_POOL_ADD_SIZE_*`) is:
  * Posix options
  * HostAP (WiFi)
  * ESP and NXP WiFi
  * ESP Bluetooth HCI driver
  * Various boards (for IPC)
  * Hawkbit (Proprietary device management?)
  * One or two other edge cases

Are those options common enough to pull in all the heap code by default for Zbus users? I would argue not.

Reply:
I guess my point is that you say "Any usual application should already use the heap" but as far as I can tell, the heap is more the exception than the rule for Zephyr applications.

Reply:
This is how I perceived it too, many areas define a number of static resources through a Kconfig symbol.